src/utils.rs

```diff
@@ -0,0 +1,280 @@
+use crate::error::SlateDBError;
+use crate::error::SlateDBError::{BackgroundTaskFailed, BackgroundTaskShutdown};
+use std::future::Future;
+
+pub(crate) struct WriteOnceRegister<T: Clone> {
+    rx: tokio::sync::watch::Receiver<Option<T>>,
+    tx: tokio::sync::watch::Sender<Option<T>>,
+}
+
+pub(crate) struct WriteOnceRegisterReader<T: Clone> {
+    rx: tokio::sync::watch::Receiver<Option<T>>,
+}
+
+impl<T: Clone> WriteOnceRegister<T> {
+    pub(crate) fn new() -> Self {
+        let (tx, rx) = tokio::sync::watch::channel(None);
+        Self { rx, tx }
+    }
+
+    pub(crate) fn write(&self, val: T) {
+        self.tx.send_if_modified(|v| {
```

Comment:
does tokio guarantee mutex in this method? (worried about a race condition where two successful writes happen). It wasn't obvious to me from the docs

Reply:
Yeah it would have to since the modifying function gets a `&mut T`